src/zenml/integrations/aws/secrets_managers/aws_secrets_manager.py

```diff
@@ -220,17 +220,21 @@ def _list_secrets(self, secret_name: Optional[str] = None) -> List[str]:
                 }
             )
 
-        # TODO [ENG-720]: Deal with pagination in the aws secret manager when
-        #  listing all secrets
-        # TODO [ENG-721]: take out this magic maxresults number
-        response = self.CLIENT.list_secrets(MaxResults=100, Filters=filters)
+        paginator = self.CLIENT.get_paginator('list_secrets')
```

Comment:
It might be worth taking this 'magic command' and including it as a constant at the top somewhere.